lib/pure/asynchttpserver.nim

```diff
@@ -261,6 +272,39 @@ proc processRequest(
       if request.body.len != contentLength:
         await request.respond(Http400, "Bad Request. Content-Length does not match actual.")
         return true
+  elif hasChunkedEncoding(request):
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding
+    var sizeOrData = 0
+    var bytesToRead = 0
+    request.body = ""
+
+    while true:
+      lineFut.mget.setLen(0)
+      lineFut.clean()
+      
+      # The encoding format alternates between specifying a number of bytes to read
+      # and the data to be read, of the previously specified size
+      if sizeOrData mod 2 == 0:
+        # Expect a number of chars to read
+        await client.recvLineInto(lineFut, maxLength = maxLine)
+        try:
+          bytesToRead = lineFut.mget.parseHexInt
+        except ValueError:
+          # Malformed request
+          await request.respond(Http411, ("Invalid chunked transfer encoding - " &
+                                          "chunk data size must be hex encoded"))
+          return true
+      else:
+        if bytesToRead == 0:
+          # Done reading chunked data
+          break
+
+        # Read bytesToRead and add to body
+        # Note we add +2 because the line must be terminated by \r\n
+        let chunk = await client.recv(bytesToRead + 2)
```

Comment:
This looks suspicious to me, I have a feeling that chunked encoding doesn't delimit data with \r\n. You should test with larger files to make sure they work (if you're just testing with "hello world" then you're unlikely to run into this).

Reply:
@dom96 The two sources I've found with details on this topic say that the chunks are delimited with `\r\n`, see:
* (MDN): https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding
* (Wikipedia): https://en.wikipedia.org/wiki/Chunked_transfer_encoding#Example

However, you were right to be suspicious! The code needs to discard these newlines, which I did not correctly do. I've fixed this (and added tests!) in this follow-up PR: https://github.com/nim-lang/Nim/pull/16678

I'd appreciate it if you could take a look and review that too! Thanks!